src/core/vdom/create-component.js

```diff
@@ -208,6 +208,10 @@ function destroy (vnode: MountedComponentVNode) {
   }
 }
 
+// return three possible result
+// 1. Ctor -> component constructor if resolved successfully
+// 2. undefined -> component is pending
+// 3. null -> component rejected
 function resolveAsyncComponent (
```

Comment:
Now resolveAsyncComponent will return null if component is rejected. This relies on JavaScript has two kinds of `zero value`. A more canonical solution is to wrap result in a container which has a property indicating resolution status. But it would incur more overhead.

Is current implementation good to you @yyx990803 ?